ICRCs/ICRC-7/ICRC-7.md

```diff
@@ -107,109 +188,174 @@ than this supply cap shall be rejected.
 icrc7_supply_cap : () -> (opt nat) query;
 ```
 
-### icrc7_metadata
+### icrc7_max_query_batch_size
+
+Returns the maximum batch size for batch query calls this ledger implementation supports.
+
+```candid "Methods" +=
+icrc7_max_query_batch_size : () -> (opt nat) query;
+```
+
+### icrc7_max_update_batch_size
+
+Returns the maximum number of token ids allowed for being used as input in a batch update method.
+
+```candid "Methods" +=
+icrc7_max_update_batch_size : () -> (opt nat) query;
+```
+
+### icrc7_default_take_value
+
+Returns the default parameter the ledger uses for `take` in case the parameter is `null` in paginated queries.
+
+```candid "Methods" +=
+icrc7_default_take_value : () -> (opt nat) query;
+```
+
+### icrc7_max_take_value
+
+Returns the maximum `take` value for paginated query calls this ledger implementation supports. The value applies to all paginated calls the ledger exposes.
 
-Returns the token metadata for a particular tokenId.
+```candid "Methods" +=
+icrc7_max_take_value : () -> (opt nat) query;
+```
+
+### icrc7_max_memo_size
+
+Returns the maximum size of `memo`s as supported by an implementation.
+
+```candid "Methods" +=
+icrc7_max_memo_size : () -> (opt nat) query;
+```
+
+### icrc7_token_metadata
+
+Returns the token metadata for `token_ids`, a list of token ids. Each tuple in the response vector comprises a token id as first element and the metadata corresponding to this token expressed as an optional record comprising `text` and `Value` pairs expressing the token metadata as second element. In case a token does not exist, no element corresponding to it is returned in the response. If a token does not have metadata, its associated metadata vector is the empty vector.
+
+ICRC-7 does not specify the representation of token metadata any further than that it is represented in a generic manner as a vector of `(text, Value)`-pairs. This is left to future standards, the collections, or the implementations in order to not constrain the utility and applicability of this standard.
+
+> [!NOTE]
+> Encoding of types not contained in the `Value` type SHOULD be handled according to best practices as put forth in the context of the ICRC-3 standard.
 
 ```candid "Type definitions" +=
-type Metadata = variant { Nat : nat; Int : int; Text : text; Blob : blob };
+// Generic value in accordance with ICRC-3
+type Value = variant { 
+    Blob : blob; 
+    Text : text; 
+    Nat : nat;
+    Int : int;
+    Array : vec Value; 
+    Map : vec record { text; Value }; 
+};
 ```
 
 ```candid "Methods" +=
-icrc7_metadata : (nat) -> (vec record { text; Metadata }) query;
+icrc7_token_metadata : (token_ids : vec nat)
+    -> (vec opt metadata: vec record { text; Value }) query;
```

Comment:
the candid doesn't seem to match the text above, i don't see any `token_id` in the response 🤔

Reply:
Good catch, adapted to match our simplified positional API.